provider/source/src/time_zones/convert.rs

```diff
@@ -190,54 +199,174 @@ impl From<CldrTimeZonesData<'_>> for ExemplarCitiesV1<'static> {
+                            .unwrap_or(IsoMinutesSinceEpoch::MAX);
+
+                        if local_end_time <= 0 {
+                            continue;
                         }
-                        None => panic!("Cannot find bcp47 for {key:?}."),
-                    },
-                    ZonePeriod::LocationOrSubRegion(place) => place
-                        .iter()
-                        .flat_map(move |(inner_key, location_or_subregion)| {
-                            let mut key = key.clone();
-                            key.push('/');
-                            key.push_str(inner_key);
-                            match location_or_subregion {
-                                MetaLocationOrSubRegion::Location(periods) => {
-                                    match bcp47_tzid_data.get(&key) {
-                                        Some(bcp47) => {
-                                            vec![(*bcp47, periods, meta_zone_id_data)]
-                                        }
-                                        None => panic!("Cannot find bcp47 for {key:?}."),
-                                    }
-                                }
-                                MetaLocationOrSubRegion::SubRegion(subregion) => subregion
+
+                        match zone_info.saving {
+                            Saving::NoSaving => {
+                                store_offsets(&mut data, local_end_time, zone_info.offset, 0);
+                            }
+
+                            Saving::OneOff(amount) => {
+                                store_offsets(&mut data, local_end_time, zone_info.offset, amount);
+                            }
+
+                            Saving::Multiple(ref rules) => {
+                                let mut rules = other.tzdb.rulesets[rules]
                                     .iter()
-                                    .flat_map(move |(inner_inner_key, periods)| {
-                                        let mut key = key.clone();
-                                        key.push('/');
-                                        key.push_str(inner_inner_key);
-                                        match bcp47_tzid_data.get(&key) {
-                                            Some(bcp47) => {
-                                                vec![(*bcp47, periods, meta_zone_id_data)]
-                                            }
-                                            None => panic!("Cannot find bcp47 for {key:?}."),
-                                        }
+                                    // Only want transitions into DST
+                                    .filter(|rule| rule.time_to_add != 0)
+                                    .filter(|rule| {
+                                        (1800..zone_info.end_time.map(|e| e.year()).unwrap_or(2500))
+                                            .any(|y| rule.applies_to_year(y))
```

Comment:
Nit: Please explain what this is doing. For example, I assume 1800 means 30 minutes, but what is 2500? (here and throughout the handling of `Saving::Multiple`)

Reply:
1800 and 2500 are years, I thought this was obvious from `e.year()` and `rule.applies_to_year`.

Reply:
ok, maybe a human parsing error. Still, I don't really understand what this is doing, and why you picked those two years, so please add comments.